### PR TITLE
Update agent to 1.20.3

### PR DESCRIPTION
**Pull request by yumex93**
### Testing
- [ ] Builds on Linux (`make release`)
- [ ] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [ ] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # Changelog
+## 1.20.3
```
Comment (sharanyad) [inline CHANGELOG.md]: Should we have the changelog entries in dev branch first?

Comment (yunhee-l) [reply CHANGELOG.md]: yeah I think so, this could cause mixup when merging master back to dev

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-1.20.2
```
Comment (haikuoliu) [inline VERSION]: You added an extra line here

File: CHANGELOG.md (hunk 1 above)
Comment (haikuoliu) [reply CHANGELOG.md]: I think we need to add change log when merging each PR instead of adding changelog together with release PR tho

Comment (petderek) [reply CHANGELOG.md]: Did you rebase this branch against dev?

Comment (yumex93) [reply CHANGELOG.md]: Yep. This pr has all changes that dev has now.

**Pull request merged**